google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/JsonStreamWriter.java

```diff
@@ -0,0 +1,359 @@
+    private String streamName;
+    private BigQueryWriteClient client;
+    private Table.TableSchema BQTableSchema;
+
+    private TransportChannelProvider channelProvider;
+    private CredentialsProvider credentialsProvider;
+    private BatchingSettings batchingSettings;
+    private RetrySettings retrySettings;
+    private ExecutorProvider executorProvider;
+    private String endpoint;
+
+    /**
+     * Constructor for JsonStreamWriter's Builder
+     *
+     * @param streamName name of the stream that must follow
+     *     "projects/[^/]+/datasets/[^/]+/tables/[^/]+/streams/[^/]+"
+     * @param BQTableSchema schema used to convert Json to proto messages.
+     * @param client
+     */
+    private Builder(
+        String streamName, Table.TableSchema BQTableSchema, BigQueryWriteClient client) {
+      this.streamName = streamName;
+      this.BQTableSchema = BQTableSchema;
+      this.client = client;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's TransportChannelProvider.
+     *
+     * @param channelProvider
+     * @return Builder
+     */
+    public Builder setChannelProvider(TransportChannelProvider channelProvider) {
+      this.channelProvider =
+          Preconditions.checkNotNull(channelProvider, "ChannelProvider is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's CredentialsProvider.
+     *
+     * @param credentialsProvider
+     * @return Builder
+     */
+    public Builder setCredentialsProvider(CredentialsProvider credentialsProvider) {
+      this.credentialsProvider =
+          Preconditions.checkNotNull(credentialsProvider, "CredentialsProvider is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's BatchingSettings.
+     *
+     * @param batchingSettings
+     * @return Builder
+     */
+    public Builder setBatchingSettings(BatchingSettings batchingSettings) {
+      this.batchingSettings =
+          Preconditions.checkNotNull(batchingSettings, "BatchingSettings is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's RetrySettings.
+     *
+     * @param retrySettings
+     * @return Builder
+     */
+    public Builder setRetrySettings(RetrySettings retrySettings) {
+      this.retrySettings = Preconditions.checkNotNull(retrySettings, "RetrySettings is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's ExecutorProvider.
+     *
+     * @param executorProvider
+     * @return Builder
+     */
+    public Builder setExecutorProvider(ExecutorProvider executorProvider) {
+      this.executorProvider =
+          Preconditions.checkNotNull(executorProvider, "ExecutorProvider is null.");
+      return this;
+    }
+
+    /**
+     * Setter for the underlying StreamWriter's Endpoint.
+     *
+     * @param endpoint
+     * @return Builder
+     */
+    public Builder setEndpoint(String endpoint) {
```

Comment:
what does this endpoint referring to? The BQ api address?

Reply:
Think so, StreamWriter had a setEndpointer() method so I included it.